go/pkg/log/server/property_test.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/chroma-core/chroma/go/pkg/proto/coordinatorpb"
+	"github.com/chroma-core/chroma/go/pkg/proto/logservicepb"
+	"github.com/chroma-core/chroma/go/pkg/types"
+	libs2 "github.com/chroma-core/chroma/go/shared/libs"
+	"github.com/jackc/pgx/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"pgregory.net/rapid"
+	"testing"
+	"time"
+)
+
+type ModelState struct {
+	CollectionEnumerationOffset map[types.UniqueID]int64
+	CollectionData              map[types.UniqueID][]*coordinatorpb.OperationRecord
+	CollectionCompactionOffset  map[types.UniqueID]int64
+}
+
+type LogServerTestSuite struct {
+	suite.Suite
+	logServer logservicepb.LogServiceServer
+	model     ModelState
+	t         *testing.T
+}
+
+func (suite *LogServerTestSuite) SetupSuite() {
+	ctx := context.Background()
+	connectionString, err := libs2.StartPgContainer(ctx)
+	assert.NoError(suite.t, err, "Failed to start pg container")
+	var conn *pgx.Conn
+	conn, err = libs2.NewPgConnection(ctx, connectionString)
+	assert.NoError(suite.t, err, "Failed to create new pg connection")
+	err = libs2.RunMigration(ctx, connectionString)
+	assert.NoError(suite.t, err, "Failed to run migration")
+	lr := repository.NewLogRepository(conn)
+	suite.logServer = NewLogServer(lr)
+	suite.model = ModelState{
+		CollectionData:             map[types.UniqueID][]*coordinatorpb.OperationRecord{},
+		CollectionCompactionOffset: map[types.UniqueID]int64{},
+	}
+}
+
+func (suite *LogServerTestSuite) TestRecordLogDb_PushLogs() {
+	ctx := context.Background()
+	// Generate collection ids
+	collections := make([]types.UniqueID, 10)
+	for i := 0; i < len(collections); i++ {
+		collections[i] = types.NewUniqueID()
+	}
+
+	collectionGen := rapid.Custom(func(t *rapid.T) types.UniqueID {
+		return collections[rapid.IntRange(0, len(collections)-1).Draw(t, "collection_id")]
+	})
+	recordGen := rapid.SliceOf(rapid.Custom(func(t *rapid.T) *coordinatorpb.OperationRecord {
+		data := rapid.SliceOf(rapid.Byte()).Draw(t, "record_data")
+		id := rapid.String().Draw(t, "record_id")
+		return &coordinatorpb.OperationRecord{
+			Id: id,
+			Vector: &coordinatorpb.Vector{
+				Vector: data,
+			},
+		}
+	}))
+	rapid.Check(suite.t, func(t *rapid.T) {
+		t.Repeat(map[string]func(*rapid.T){
+			"pushLogs": func(t *rapid.T) {
+				c := collectionGen.Draw(t, "collection")
+				records := recordGen.Draw(t, "record")
+				r, err := suite.logServer.PushLogs(ctx, &logservicepb.PushLogsRequest{
+					CollectionId: c.String(),
+					Records:      records,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if int32(len(records)) != r.RecordCount {
+					t.Fatal("record count mismatch", len(records), r.RecordCount)
+				}
+				suite.model.CollectionData[c] = append(suite.model.CollectionData[c], records...)
+			},
+			"getAllCollectionsToCompact": func(t *rapid.T) {
+				result, err := suite.logServer.GetAllCollectionInfoToCompact(ctx, &logservicepb.GetAllCollectionInfoToCompactRequest{})
```

Comment:
Can we add more sanity check for getAllCollectionsToCompact? One check could be the checking the number of collections to compact matches the number of collections in the model.